internal/git/gitsync_processor.go

```diff
@@ -127,15 +156,243 @@ func watchRepo(ctx context.Context, restConfig *rest.Config, repo *v1.Repository
 		logger.Debug("no monitoring")
 
 	} else {
-		// TODO: monitoring with intervals
-		logger.Debug("monitoring with intervals")
+		ticker := time.NewTicker(timeInterval * time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				patchedResources, err := CheckForRepoUpdates(r, repo, &gitSync.Status, ctx)
+				if err != nil {
+					return err
+				}
+
+				for key, afterValue := range patchedResources.After {
+					namespace := strings.Split(key, "-")[0]
```

Comment:
so what should we ideally use to create the key for map ,how should we do the combination of namespace and name or should we keep the name only ?

Reply:
I'm actually thinking a map of namespace to map where key is name should be faster than a single map with a delimiter, since with the delimiter we have to do an O(n) search each time. (Sorry, I know the original issue didn't mention the double map, but I think it would be faster.)

Reply:
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names Per the naming restriction here. I think we can use other delimiter like '/'

Reply:
@xdevxy we will have to do a search on the string each time here for every resource in a file that changed if we have a single level map. Is your thought that it's not a big impact, and are you preferring to get this merged sooner than later? If so, I can see that point of view too.

Reply:
okay, never mind the 2 level map. Git commits are pretty infrequent, so I don't think we need to worry about that level of performance. We can just change the delimiter to "/"